service/src/main/java/bio/terra/cbas/dependencies/wds/ScheduledWdsClientRefresher.java

```diff
@@ -0,0 +1,88 @@
+package bio.terra.cbas.dependencies.wds;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import javax.ws.rs.client.Client;
+import org.databiosphere.workspacedata.client.ApiClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ScheduledWdsClientRefresher {
+
+  private final Object synchronizer = new Object();
+
+  private final Logger logger = LoggerFactory.getLogger(ScheduledWdsClientRefresher.class);
+
+  private final List<Client> retiredClients;
+  private final List<Client> expiredClients;
+
+  private Client currentClient;
```

Comment:
Oops, I see you have already made the scheduled tasks synchronized. That's good but I think you still have an issue with the getters since they're not synchronized. Maybe making a copy of the lists before returning them would be sufficient.

Reply:
The lists are not intended to be accessed outside the class. Good point on adding synchronization around the `getCurrentClient`... though I'm not sure it would actually be possible for it to ever be in an invalid state.